calcom/cal.com
# fix: use a dedicated shadow database for migrations

Use a dedicated shadow database for migrations, needed when the db user cannot create a database, for example when using cloud providers

## What does this PR do?

This PR fixes an issues people might encounter when running migrations and the database user cannot create a new shadow database. This is the case when using cloud providers for example. 

The solutions is simply to create a second database, and configure to be used by Prisma. 

This needed to add a new optionnal ENV variable. 



## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

- Try to run the database migration with a database that cannot CREATE DATABASE
- Expect it to fail
- Create a new database and add needed connection information  to .env new variable `SHADOW_DATABASE_URL`
- Run the database migration
- Expect it to success

## Checklist

- OK
